server/app/auth/CiviFormProfileData.java

```diff
@@ -41,8 +43,13 @@ public CiviFormProfileData() {
   }
 
   public CiviFormProfileData(Long accountId) {
+    this(accountId, Clock.systemUTC());
```

Comment:
(blocking) I don't like how we're using mixed version of the clock. Internal to this class clock is being used as method parameters, a constructor parameter, use of static Clock. This class isn't being created by Guice so we can't just `@Inject` it, but there needs to be one entry point of the clock into this class.

The Clock typically provided when injecting it into the constructor used the timezone that is configured by CiviForm admins. Having each method accept the clock may result in mismatched clocks being used when setting or comparing times.

Please adjust this so that there is only a single clock that is used in this class.

Reply:
This constructor was only used for tests. I have removed this constructor and now we only use the clock passed to the profile data.